tests/test_cases/test_iteration_verilog/test_iteration_es.py

```diff
@@ -52,25 +53,34 @@ async def recursive_discovery(dut):
     else:
         pass_total = 265
 
+    # Icarus doesn't support array indexes like get_handle_by_name("some_path[x]")
+    SKIP_HANDLE_ASSERT = cocotb.SIM_NAME.lower().startswith(("riviera", "icarus"))
+
     tlog = logging.getLogger("cocotb.test")
 
-    def dump_all_the_things(parent):
+    def sim_iter(parent):
         if not isinstance(
             parent,
             (
                 cocotb.handle.RegionObject,
                 cocotb.handle.NonHierarchyIndexableObjectBase,
             ),
         ):
-            return 0
-        count = 0
+            return
         for thing in parent:
-            count += 1
-            tlog.info("Found %s (%s)", thing._path, type(thing))
-            count += dump_all_the_things(thing)
-        return count
+            yield thing
+            yield from sim_iter(thing)
+
+    total = 0
+    for thing in sim_iter(dut):
+        tlog.info("Found %s (%s)", thing._path, type(thing))
+
+        subpath = thing._path.split(".", 1)[1]
+        if not SKIP_HANDLE_ASSERT:
+            assert dut._handle.get_handle_by_name(subpath) == thing._handle
```

Comment:
So just to get this straight, this is testing that the object retrieved by name from the root object is the same object as the one found during iteration-based discovery? And that is not the case for Icarus and Riviera for some reasons... A comment might be worthwhile.

The above C code changes look good.

Reply:
Yeah, but I might've mistaked the CI being out of commission as a test failure?

Reply:
Well it was just re-run and seems to be working now.

Reply:
ok I'll revert and push to see if it works on those sims